### PR TITLE
chore: updated dependencies removed unused dependencies.

### LINKED ISSUE #19448: [Task]: Update json v5 library
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Update json v5 library

### PR DESCRIPTION
**Pull request by yaldram**
## Description

- Add resolution for json5 dependency to resolve package vulnerabilities.
- Remove babel-plugin-styled-components because it was unused.
- Related to `react-scripts` , if everything builds and dev server is running as expected we have no issues.

Fixes #19448

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

> Please delete options that are not relevant.

- Non-breaking change.

- Manual
- Jest
- Cypress

### Test Plan
- All tests need to pass.
- Does not need a regression.

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by riodeuno** [comment]
@yaldram Thank you for cleaning up the codebase. I really appreciate this.

There are a few things I'd like to ask you to add to the PR description.
1. What was the issue?
2. What changes have we made to resolve the issue?
3. The task issue id which calls for this resolution.
4. How do we know if this change breaks anything? (Which tests would tell us this)

**Review by riodeuno — COMMENTED** [review]
The changes look good to me. I have requests to update the PR description in a previous comment.

**Comment by yaldram** [comment]
@riodeuno I have updated the pull request description. This Pull request resolves a vulnerability, it needs no Q/A as it is related to the react-scrips dependencies, we aren't shipping this code to our customers.

**Comment by riodeuno** [comment]
@yaldram I don't see a few details in the PR:
1. An issue id.
2. The PR seems to say Breaking change. If this is correct, we need to elaborate on what could break.
3. If we only expect existing tests to pass, we need to mention that in the tests section.
4. If we need a full regression for this, we will need to mention that as well.

**Comment by yaldram** [comment]
@riodeuno done.

**Review by riodeuno — APPROVED** [review]

File: app/client/package.json [hunk 1]
```diff
@@ -302,8 +301,8 @@
     "focus-trap-react/**/tabbable": "5.2.1",
     "json-schema": "0.4.0",
     "node-fetch": "2.6.7",
-    "babel-plugin-styled-components": "2.0.7",
     "minimatch": "^5.0.0",
-    "loader-utils": "^2.0.4"
+    "loader-utils": "^2.0.4",
+    "json5": "2.2.2"
```
Comment (IAmAnubhavSaini) [inline app/client/package.json:306]: @yaldram version 2.2.3 is available now, please use that one.

Comment (yaldram) [reply app/client/package.json]: Done, thanks

**Review by IAmAnubhavSaini — APPROVED** [review]

**Pull request merged**